src/cursor/cur_layered.c

```diff
@@ -1426,15 +1426,15 @@ __clayered_remove_int(
     } else {
         c = clayered->ingest_cursor;
         clayered->current_cursor = c;
-        if (!positioned) {
+        /* We may be positioned on the stable table so check again. */
+        if (!F_ISSET(c, WT_CURSTD_KEY_INT)) {
```

Comment:
@jiechenbo I think this may still not be the 100% correct fix. For the case that the ingest table is positioned but not positioned on the correct key, we should set the key as well. @jiechenbo Can you create a ticket to fix this?